runtime/compiler/trj9/runtime/RelocationRecord.cpp

```diff
@@ -293,162 +296,172 @@ TR_RelocationRecordGroup::handleRelocation(TR_RelocationRuntime *reloRuntime,
          break;
       case TR_ClassAddress:
-         new (this) TR_RelocationRecordClassObject();
+         reloRecord = new (storage) TR_RelocationRecordClassObject();
          break;
       case TR_MethodObject:
-         new (this) TR_RelocationRecordMethodObject();
+         reloRecord = new (storage) TR_RelocationRecordMethodObject();
          break;
       case TR_PicTrampolines:
-         new (this) TR_RelocationRecordPicTrampolines();
+         reloRecord = new (storage) TR_RelocationRecordPicTrampolines();
          break;
       case TR_Trampolines:
-         new (this) TR_RelocationRecordTrampolines();
+         reloRecord = new (storage) TR_RelocationRecordTrampolines();
          break;
       case TR_Thunks:
-         new (this) TR_RelocationRecordThunks();
+         reloRecord = new (storage) TR_RelocationRecordThunks();
          break;
       case TR_GlobalValue:
-         new (this) TR_RelocationRecordGlobalValue();
+         reloRecord = new (storage) TR_RelocationRecordGlobalValue();
          break;
       case TR_BodyInfoAddressLoad:
-         new (this) TR_RelocationRecordBodyInfoLoad();
+         reloRecord = new (storage) TR_RelocationRecordBodyInfoLoad();
          break;
       case TR_ArrayCopyHelper:
-         new (this) TR_RelocationRecordArrayCopyHelper();
+         reloRecord = new (storage) TR_RelocationRecordArrayCopyHelper();
          break;
       case TR_ArrayCopyToc:
-         new (this) TR_RelocationRecordArrayCopyToc();
+         reloRecord = new (storage) TR_RelocationRecordArrayCopyToc();
          break;
       case TR_RamMethodSequence:
       case TR_RamMethodSequenceReg:
-         new (this) TR_RelocationRecordRamSequence();
+         reloRecord = new (storage) TR_RelocationRecordRamSequence();
          break;
       case TR_FixedSequenceAddress:
       case TR_FixedSequenceAddress2:
-         new (this) TR_RelocationRecordWithOffset();
+         reloRecord = new (storage) TR_RelocationRecordWithOffset();
          break;
       case TR_HCR:
-         new (this) TR_RelocationRecordHCR();
+         reloRecord = new (storage) TR_RelocationRecordHCR();
          break;
       case TR_ClassPointer:
-         new (this) TR_RelocationRecordClassPointer();
+         reloRecord = new (storage) TR_RelocationRecordClassPointer();
          break;
       case TR_MethodPointer:
-         new (this) TR_RelocationRecordMethodPointer();
+         reloRecord = new (storage) TR_RelocationRecordMethodPointer();
          break;
       case TR_EmitClass:
-         new (this) TR_RelocationRecordEmitClass();
+         reloRecord = new (storage) TR_RelocationRecordEmitClass();
          break;
       default:
          // TODO: error condition
          printf("Unexpected relo record: %d\n", reloType);fflush(stdout);
          TR_ASSERT(0, "Unexpected relocation record type found");
          exit(0);
       }
+      reloRecord->_reloRuntime = reloRuntime;
```

Comment:
It's more code to change, but adding these fields to the constructors for all the relocation classes (which pass them into the base class constructor) to initialize would be a lot cleaner. Then you don't need to add the "setBinaryRecord()" function which isn't really something RelocationRecord should be allowing (especially as a public function).